code/modules/antagonists/ninja/weapons/energy_glaive.dm

```diff
@@ -0,0 +1,110 @@
+/**
+ * # Energy Glaive
+ *
+ * The space ninja's glaive.
+ *
+ * The glaive that only space ninja spawns with.  Comes with 12 force and 40 throwforce, along with a throw-teleport system.
+ * Throwing glaive will teleport thrower to where it arrived. Knockdown people that stay on turf where user teleport.
+ * 4 charges witch 10 SECONDS cooldown.
+ * Make two attack instead of one. Target will be chosen from any living mob in 3x3 aoe of user.
+ * If no one in range glaive will make second attack to first target again.
+ *
+ */
+#define GLAIVE_COOLDOWN 10 SECONDS
+/obj/item/energy_glaive
+	name = "energy glaive"
+	desc = "Double-sided energy spear."
+	desc_controls = "Throw it to teleport."
+	icon = 'icons/obj/weapons/sword.dmi'
+	icon_state = "glaive"
+	inhand_icon_state = "glaive"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	force = 12 // 12 + 12 = 24
+	throwforce = 40
+	block_chance = 70
+	armour_penetration = 50
+	w_class = WEIGHT_CLASS_NORMAL
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	pickup_sound = 'sound/items/unsheath.ogg'
+	drop_sound = 'sound/items/sheath.ogg'
+	block_sound = 'sound/weapons/block_blade.ogg'
+	attack_verb_continuous = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "cuts")
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
+	sharpness = SHARP_EDGED
+	max_integrity = 200
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	var/datum/effect_system/spark_spread/spark_system
+	/// If if double attacked other target we don't attack first target again
```

Comment:
```suggestion
	/// Tracks if the next attack will be the automatic double attack to prevent an infinite attacking loop.```

Reply:
if there are no other targets near the attacker, then he attacks the same target again
Priority like:
1. You attacked fist target
2. Try to find second target
3. Attack first target if we haven't found second target